Gems/Multiplayer/Code/Tests/ServerHierarchyBenchmarks.cpp

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#ifdef HAVE_BENCHMARK
+#include <CommonBenchmarkSetup.h>
+#include <AzCore/std/smart_ptr/make_shared.h>
+
+namespace Multiplayer
+{
+    /*
+     * Hierarchy of 16 entities: Parent -> Child_2 -> .. -> Child_16
+     */
+    class ServerDeepHierarchyBenchmark : public HierarchyBenchmarkBase
+    {
+    public:
+        const NetEntityId RootNetEntityId = NetEntityId{ 1 };
+        const NetEntityId ChildNetEntityId = NetEntityId{ 2 };
+        const NetEntityId ChildOfChildNetEntityId = NetEntityId{ 3 };
+
+        void internalSetUp() override
+        {
+            HierarchyBenchmarkBase::internalSetUp();
+
+            m_root = AZStd::make_unique<EntityInfo>((1), "root", RootNetEntityId, EntityInfo::Role::Root);
+            CreateParent(*m_root);
+
+            m_children = AZStd::make_unique<AZStd::vector<AZStd::shared_ptr<EntityInfo>>>();
+
+            EntityInfo* parent = m_root.get();
+            for (int i = 0; i < 15; ++i)
```

Comment:
nit: I would call out the requirement that 16 is a deep as hierarchy can go now, its not clear to future maintainers why 15 is the stopping condition here.

Reply:
Sounds good, will do.